test/functional/p2p_addrv2_relay.py

```diff
@@ -74,6 +74,16 @@ def run_test(self):
 
         assert addr_receiver.addrv2_received_and_checked
 
+        self.log.info("Check that the node sends a SENDADDRV2 message to outbound full relay peers")
+        conn = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=0)
+        conn.sync_with_ping()
+        assert_equal(conn.message_count['sendaddrv2'], 1)
+
+        self.log.info("Check that the node does not send a SENDADDRV2 message to outbound block-relay-only peers")
+        conn = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, connection_type="block-relay-only")
+        conn.sync_with_ping()
```

Comment:
question in ee5fb1f8dcdca66af992f05081c71703aa43fdff:

Is there a reason for the sync_with_ping's? Ideally the `add_outbound_p2p_connection` helper should fully setup the connection (including handshake). And it seems it does do that, as there is a `p2p_conn.wait_for_verack()` and `p2p_conn.sync_with_ping()` inside.

Reply:
ah, agree its not necessary. I missed that there's already a `sync_with_ping()` inside `add_outbound_p2p_connection`, so I was adding buffer.

Reply:
removed